src/tree/updater_colmaker.cc

```diff
@@ -157,11 +157,11 @@ class ColMaker: public TreeUpdater {
             feat_index.push_back(i);
           }
         }
-        unsigned n = static_cast<unsigned>(param.colsample_bytree * feat_index.size());
+        unsigned n = std::max(static_cast<unsigned>(1),
+                              static_cast<unsigned>(param.colsample_bytree * feat_index.size()));
```

Comment:
@tqchen is this change ok? We have experienced crashes during grid search of parameters when n==0.